### DISCUSSION TITLE
any plan support idl file?

### DISCUSSION THREAD

**Discussion by romanholidaypancakes** [stephenberry/glaze#20] [Ideas]
automatically output the corresponding structure according to idl,no need to manually enter the following code

```
template <>
struct glz::meta<my_struct> {
   using T = my_struct;
   static constexpr auto value = object(
      "i", &T::i,
      "d", &T::d,
      "hello", &T::hello,
      "arr", &T::arr
   );
```

**Comment by stephenberry** [comment]
idl files are not currently on our roadmap, but it is an interesting proposition. I'll keep it in mind, but it probably won't happen soon unless someone wants to contribute or the team I lead finds a need for it.

**Reply by stephenberry** [reply]
I'll note that this entire meta structure will not be needed once we have reflection in C++, we'll be able to automatically generate it.
